src/main/java/com/linecorp/armeria/client/ClientBuilder.java

```diff
@@ -249,13 +252,21 @@ public ClientBuilder codecDecorator(
     private Client newClient(Class<?> interfaceClass) {
         final Scheme scheme = Scheme.parse(uri.getScheme());
         final SessionProtocol sessionProtocol = scheme.sessionProtocol();
+        URI codecURI = this.uri;
 
-        final RemoteInvoker remoteInvoker = remoteInvokerFactory.getInvoker(sessionProtocol);
+        RemoteInvoker remoteInvoker = remoteInvokerFactory.getInvoker(sessionProtocol);
         if (remoteInvoker == null) {
             throw new IllegalArgumentException("unsupported scheme: " + scheme);
         }
 
-        final ClientCodec codec = createCodec(uri, scheme, interfaceClass);
+        final String groupName = EndpointGroupUtil.getEndpointGroupName(this.uri);
+        if (groupName != null) {
+            remoteInvoker = new EndpointGroupInvoker(remoteInvoker, groupName);
+            //for RFC2616
```

Comment:
// Ensure a valid URL is returned in business logic that accesses it through the ServiceInvocationContext by stripping out our custom markers.